test/extended/builds/cluster_config.go

```diff
@@ -107,36 +230,378 @@ var _ = g.Describe("[Feature:Builds][Serial][Slow][Disruptive] alter builds via
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(buildConfig.Spec.BuildDefaults.DefaultProxy).NotTo(o.BeNil())
+				g.By("starting build simple-s2i-build and waiting for failure")
+				br, err := exutil.StartBuildAndWait(oc, "simple-s2i-build")
+				o.Expect(err).NotTo(o.HaveOccurred())
+				br.AssertFailure()
+				g.By("expecting the build logs to indicate invalid proxy")
+				buildLog, err := br.LogsNoTimestamp()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(buildLog).To(o.ContainSubstring("Could not resolve proxy: invalid.proxy.redhat.com; Unknown error"))
+				g.By("checking pod as well")
+				pod, err := oc.KubeClient().CoreV1().Pods(oc.Namespace()).Get(br.BuildName+"-build", metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				checkPodProxyEnvs(pod.Spec.Containers, buildConfig.Spec.BuildDefaults.DefaultProxy)
+				checkPodProxyEnvs(pod.Spec.InitContainers, buildConfig.Spec.BuildDefaults.DefaultProxy)
+			})
+
+			// this replaces coverage from the TestBuildDefaultGitHTTPProxy and TestBuildDefaultGitHTTPSProxy integration test
+			g.It("Apply git proxy configuration to build pod", func() {
+				g.By("apply proxy cluster configuration")
+				buildConfig, err := oc.AdminConfigClient().ConfigV1().Builds().Get("cluster", metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				buildConfig.Spec.BuildDefaults.GitProxy = &configv1.ProxySpec{
+					HTTPProxy:  "http://invalid.proxy.redhat.com:3288",
+					HTTPSProxy: "https://invalid.proxy.redhat.com:3288",
+					NoProxy:    "image-registry.openshift-image-registry.svc:5000",
+				}
+				buildConfig, err = oc.AdminConfigClient().ConfigV1().Builds().Update(buildConfig)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				g.By("waiting 10s for build controller to detect proxy cfg chg")
 				time.Sleep(10 * time.Second)
-				g.By("starting build simple-docker-build and waiting for failure")
-				br, err := exutil.StartBuildAndWait(oc, "simple-docker-build")
+				g.By("verify build.config is set")
+				buildConfig, err = oc.AdminConfigClient().ConfigV1().Builds().Get("cluster", metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(buildConfig.Spec.BuildDefaults.GitProxy).NotTo(o.BeNil())
+
+				g.By("starting build simple-s2i-build and waiting for failure")
+				br, err := exutil.StartBuildAndWait(oc, "simple-s2i-build")
 				o.Expect(err).NotTo(o.HaveOccurred())
 				br.AssertFailure()
 				g.By("expecting the build logs to indicate invalid proxy")
 				buildLog, err := br.LogsNoTimestamp()
 				o.Expect(err).NotTo(o.HaveOccurred())
 				o.Expect(buildLog).To(o.ContainSubstring("Could not resolve proxy: invalid.proxy.redhat.com; Unknown error"))
+				g.By("checking build stored in pod as well")
+				// note, only the build stored in the Pod's "BUILD" env var has the updated proxy settings; they do not
+				// get propagated to the associated build stored in etcd
```

Comment:
Correct - I believe this has been identified as a long-term thing to fix (ex: by augmenting the `status` sub-resource for builds)